services/galley/src/Galley/API/Federation.hs

```diff
@@ -67,3 +76,15 @@ updateConversationMemberships cmu = do
   let targets = nubOrd $ cmuAlreadyPresentUsers cmu <> localUserIds
   -- FUTUREWORK: support bots?
   pushConversationEvent event targets []
+
+-- FUTUREWORK: report errors to the originating backend
+receiveMessage :: Domain -> RemoteMessage -> Galley ()
+receiveMessage domain rm =
+  API.postRemoteToLocal
+    (rmTime rm)
+    (Tagged (Qualified (rmConversation rm) domain))
+    (fmap (,(rmSenderClient rm)) (rmSender rm))
```

Comment:
Why create a tuple here?

Reply:
I thought having the `Qualified` wrapper around both `UserId` and `ClientId` made it clearer that they belong together. But really there's no serious reason, I can just split them into a `Qualified UserId` and a `ClientId` if you prefer.